File: trueface-sdk-android/sdk/src/main/java/ai/trueface/sdk/models/FaceDetection.java
package ai.trueface.sdk.models;

import com.google.gson.annotations.Expose;
import com.google.gson.annotations.SerializedName;

import java.util.List;

public class FaceDetection {

    @SerializedName("message")
    @Expose
    private String message;
    @SerializedName("success")
    @Expose
    private Boolean success;
    @SerializedName("faces")
    @Expose
    private List<FaceDetectionFace> faces = null;

    public String getMessage() {
        return message;
    }

    public void setMessage(String message) {
        this.message = message;
    }

    public Boolean getSuccess() {
        return success;
    }

    public void setSuccess(Boolean success) {
        this.success = success;
    }

    public List<FaceDetectionFace> getFaces() {
        return faces;
    }

    public void setFaces(List<FaceDetectionFace> faces) {
        this.faces = faces;
    }

}

File: trueface-sdk-android/sdk/src/main/java/ai/trueface/sdk/models/FaceDetection.kt
package ai.trueface.sdk.models

import android.os.Parcel
import android.os.Parcelable
import com.google.gson.annotations.Expose
import com.google.gson.annotations.SerializedName

class FaceDetection() : Parcelable {

    @SerializedName("message")
    @Expose
    var message: String? = null

    @SerializedName("success")
    @Expose
    var success: Boolean = false

    @SerializedName("faces")
    @Expose
    var faces: List<Face>? = null

    constructor(parcel: Parcel) : this() {
        message = parcel.readString()
        success = parcel.readValue(Boolean::class.java.classLoader) as Boolean
    }

    override fun writeToParcel(parcel: Parcel, flags: Int) {
        parcel.writeString(message)
        parcel.writeValue(success)
    }

    override fun describeContents(): Int {
        return 0
    }

    companion object CREATOR : Parcelable.Creator<FaceDetection> {
        override fun createFromParcel(parcel: Parcel): FaceDetection {
            return FaceDetection(parcel)
        }

        override fun newArray(size: Int): Array<FaceDetection?> {
            return arrayOfNulls(size)
        }
    }

}
